table/block_based/block_based_table_builder.cc

```diff
@@ -369,6 +367,33 @@ struct BlockBasedTableBuilder::Rep {
   uint64_t get_offset() { return offset.load(std::memory_order_relaxed); }
   void set_offset(uint64_t o) { offset.store(o, std::memory_order_relaxed); }
 
+  bool ok() const {
+    assert(io_status.ok() || !status.ok());
+    return status.ok();
+  }
+
+  const IOStatus& get_io_status() const { return io_status; }
+
+  const Status& get_status() const { return status; }
+
+  void GetStatusFromIOStatus() {
```

Comment:
do you mean `SetStatusFromIOStatus()`? or maybe `SyncStatusFromIOStatus()` is clearer. Or even drop it entirely and let `get_status()` return the `io_status` when `status.ok()`.

Reply:
Sure let me do that.